code/game/objects/structures/flora.dm

```diff
@@ -1,48 +1,222 @@
-//trees
+/obj/structure/flora
+	anchored = TRUE
+	var/health = 25
+	var/health_max = 25
+	var/on_fire = FALSE
+
+/obj/structure/flora/ex_act(severity)
+	switch(severity)
+		if(1)
+			cdel(src)
+		if(2)
+			if(prob(70))
+				cdel(src)
+		if(3)
+			if(prob(50))
+				cdel(src)
+
+/obj/structure/flora/Dispose()
+	processing_objects.Remove(src)
+	return ..()
+
+/obj/structure/flora/attackby(obj/item/W, mob/living/user)
+	if(!W || !user || (W.flags_item & NOBLUDGEON))
+		return FALSE
+
+	var/damage = W.force
+	if(W.w_class < 3 || !W.sharp || W.force < 20) //only big strong sharp weapon are adequate
+		damage *= 0.25
+
+	if(istype(W, /obj/item/tool/weldingtool))
+		var/obj/item/tool/weldingtool/WT = W
+
+		if(WT.remove_fuel(5))
+			damage = 25
+			playsound(loc, 'sound/items/Welder.ogg', 25, 1)
+
+	else
+		playsound(loc, "alien_resin_break", 25)
+	user.animation_attack_on(src)
+
+	var/multiplier = 1
+	if(W.damtype == "fire") //Burn damage deals extra
+		multiplier += 1
+
+/obj/structure/flora/update_icon()
+	return
+
+/obj/structure/flora/flamer_fire_act()
+	if(on_fire)
+		return
+	on_fire = TRUE
+	processing_objects.Add(src)
+
+/obj/structure/flora/process()
+	if(health <= 0)
+		cdel(src)
+	if(!on_fire)
+		processing_objects.Remove(src)
+	else
+		health -= 25
+
+//TREES
+
 /obj/structure/flora/tree
 	name = "tree"
-	anchored = 1
-	density = 1
+	desc = "A large tree."
+	density = TRUE
 	pixel_x = -16
+	health = 500
+	health_max = 500
 	layer = ABOVE_FLY_LAYER
+	var/log_amount = 10
+
+/obj/structure/flora/tree/ex_act(severity)
+	switch(severity)
+		if(1.0)
+			health -= 500
+		if(2.0)
+			health -= (rand(140, 300))
+		if(3.0)
+			health -= (rand(50, 100))
+	processing_objects.Add(src)
+	return
+
+/obj/structure/flora/tree/bullet_act(var/obj/item/projectile/Proj)
+	health -= Proj.damage * 0.5
+	. = ..()
+	processing_objects.Add(src)
+	return TRUE
+
+/obj/structure/flora/tree/attackby(obj/item/W, mob/user, params)
+	if(W.sharp && W.force > 0)
+		if(W.hitsound)
+			playsound(get_turf(src), W.hitsound, 100, 0, 0)
+		user.visible_message("<span class='notice'>[user] begins to cut down [src] with [W].</span>","<span class='notice'>You begin to cut down [src] with [W].</span>", "You hear the sound of sawing.")
+		if(!do_after(usr, 100, TRUE, 5, BUSY_ICON_BUILD)) //5 seconds with 20 force, 8 seconds with a hatchet, 20 seconds with a shard.
```

Comment:
The description is not correct here, is it? I seems to be 10 seconds regardless of force/item.